bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/src/main/java/com/google/cloud/bigtable/mapreduce/hbasesnapshots/ImportHBaseSnapshotJob.java

```diff
@@ -0,0 +1,317 @@
+   */
+  private static void usage(final String errorMsg) {
+    if (errorMsg != null && errorMsg.length() > 0) {
+      System.out.println("ERROR: " + errorMsg);
+    }
+    System.out.println(
+        "Usage: hadoop jar <JAR> <CLASS> <properties> <snapshot-name> <snapshot-dir> <tablename> <temp-dir>\n"
+            + "Required properties: \n"
+            + "  -D"
+            + BigtableOptionsFactory.PROJECT_ID_KEY
+            + "=<bigtable-project-id>\n"
+            + "  -D"
+            + BigtableOptionsFactory.INSTANCE_ID_KEY
+            + "=<bigtable-instance-id>\n"
+            + "Required arguments:\n"
+            + "  snapshot-name     - name of hbase snapshot to read\n"
+            + "  snapshot-dir      - directory of snapshot, e.g., gs://hbase-migration-table1-bucket/export/table1-snapshot\n"
+            + "  tablename         - bigtable table to import into\n"
+            + "  temp-dir          - a temporary base directory to restore hlinks of the snapshot into for read, e.g., gs://hbase-migration-table1-bucket/export/table1-restore\n"
+            + "Optional properties: \n"
+            + "  -D"
+            + BigtableOptionsFactory.BIGTABLE_BUFFERED_MUTATOR_ENABLE_THROTTLING
+            + "=(true|false)\n"
+            + "  -D"
+            + BigtableOptionsFactory.BIGTABLE_BUFFERED_MUTATOR_THROTTLING_THRESHOLD_MILLIS
+            + "=<throttling-threshold-ms>\n");
+
+    System.exit(1);
+  }
+
+  /**
+   * Job runner
+   *
+   * @param args command line input
+   * @return job success flag
+   * @throws Exception terminal for application failure
+   */
+  @Override
+  public int run(String[] args) throws Exception {
+    Configuration conf = HBaseConfiguration.create(getConf());
+    String[] otherArgs = new GenericOptionsParser(conf, args).getRemainingArgs();
+
+    if (setConfFromArgs(conf, otherArgs) != 0) return -1;
+
+    LOG.info(
+        "Using Arguments: "
+            + "\n"
+            + "snapshotName = "
+            + conf.get(SNAPSHOTNAME_KEY)
+            + "\n"
+            + "snapshotDir = "
+            + conf.get(HConstants.HBASE_DIR)
+            + "\n"
+            + "bigtableName = "
+            + conf.get(TableOutputFormat.OUTPUT_TABLE)
+            + "\n"
+            + "tempDir = "
+            + conf.get(SNAPSHOT_RESTOREDIR_KEY)
+            + "\n"
+            + CommonConfigurationKeysPublic.FS_DEFAULT_NAME_KEY
+            + " = "
+            + conf.get(CommonConfigurationKeysPublic.FS_DEFAULT_NAME_KEY)
+            + "\n");
+
+    Job job = createSubmittableJob(conf);
+    int successStatus = (job.waitForCompletion(true) ? 0 : 1);
+
+    Counters counters = job.getCounters();
+    long numRows = counters.findCounter(ScanCounter.NUM_ROWS).getValue();
+    long numCells = counters.findCounter(ScanCounter.NUM_CELLS).getValue();
+
+    System.out.println(("############# Num of ROWS imported= " + numRows));
+    System.out.println(("############# Num of CELLS imported= " + numCells));
```

Comment:
should these be logs? it seems inconsistent to log the start of the job but sout the end

Reply:
done